test/flavors/custom-element/analyze-built-ins-test.ts

```diff
@@ -0,0 +1,38 @@
+import { join } from "path";
+import { analyzeSourceFile } from "../../../src/analyze/analyze-source-file";
+import { getCurrentTsModule, getCurrentTsModuleDirectory, tsTest } from "../../helpers/ts-test";
+
+tsTest("analyzeSourceFile on lib.dom.ts returns correct result", t => {
+	const tsModule = getCurrentTsModule();
+	const program = tsModule.createProgram([join(getCurrentTsModuleDirectory(), "lib.dom.d.ts")], {});
+
+	const endsWithLibDom = "lib.dom.d.ts";
+
+	const domLibSourceFile = program.getSourceFiles().find(sf => sf.fileName.endsWith(endsWithLibDom));
+	if (domLibSourceFile == null) {
+		throw new Error(`Couldn't find '${endsWithLibDom}'. Have you included the 'dom' lib in your tsconfig?`);
+	}
+
+	const result = analyzeSourceFile(domLibSourceFile, {
+		program,
+		ts: tsModule,
+		config: {
+			features: ["event", "member", "slot", "csspart", "cssproperty"],
+			analyzeGlobalFeatures: false, // Don't analyze global features in lib.dom.d.ts
+			analyzeDefaultLib: true,
+			analyzeDependencies: true,
+			analyzeAllDeclarations: false,
+			excludedDeclarationNames: ["HTMLElement"]
```

Comment:
Why exclude HTMLElement?

Reply:
I'm not sure, but these are the config options used in lit-analyzer and I wanted a regression test against that case which showed the problem with built-in element properties.